docs/src/index.md

```diff
@@ -362,6 +362,116 @@ HDF5 is a large library, and the low-level wrap is not complete. However, many o
+`Pkg.build("HDF5")`.
+This is in particular needed for [parallel HDF5 support](@ref Parallel-HDF5),
+which is not provided by the `HDF5_jll` binaries.
+
+For example, you can set `JULIA_HDF5_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/hdf5/mpich/`
+if you're using the system package [`libhdf5-mpich-dev`](https://packages.ubuntu.com/focal/libhdf5-mpich-dev)
+on Ubuntu 20.04.
+
+
+## Parallel HDF5
+
+It is possible to read and write [parallel
+HDF5](https://portal.hdfgroup.org/display/HDF5/Parallel+HDF5) files using MPI.
+For this, the HDF5 binaries loaded by HDF5.jl must have been compiled with
+parallel support, and linked to the specific MPI implementation that will be used for parallel I/O.
+
+Parallel-enabled HDF5 libraries are usually included in computing clusters and
+linked to the available MPI implementations.
+They are also available via the package manager of a number of Linux
+distributions.
+
+Finally, note that the MPI.jl package is lazy-loaded by HDF5.jl
+using [Requires](https://github.com/JuliaPackaging/Requires.jl).
+In practice, this means that in Julia code, `MPI` must be imported *before*
+`HDF5` for parallel functionality to be available.
+
+### Setting-up Parallel HDF5
+
+The following step-by-step guide assumes one already has access to
+parallel-enabled HDF5 libraries linked to an existent MPI installation.
+
+#### 1. Using system-provided MPI libraries
+
+Set the environment variable `JULIA_MPI_BINARY=system` and then run
+`]build MPI` from Julia.
+For more control, one can also set the `JULIA_MPI_PATH` environment variable
+to the top-level installation directory of the MPI library.
+See the [MPI.jl
+docs](https://juliaparallel.github.io/MPI.jl/stable/configuration/#Using-a-system-provided-MPI-1)
+for details.
+
+#### 2. Using parallel HDF5 libraries
+
+As detailed in [System-provided HDF5 libraries](@ref), set the
+`JULIA_HDF5_LIBRARY_PATH` environment variable to the directory where
+the HDF5 libraries compiled with parallel support are found.
+
+#### 3. Loading MPI-enabled HDF5
+
+In Julia code, MPI.jl must be loaded *before* HDF5.jl for MPI functionality to
+be available:
+
+```julia
+using MPI
+using HDF5
+```
+
+### Reading and writing data in parallel
+
+A parallel HDF5 file may be opened by passing a `MPI.Comm` (and optionally a
+`MPI.Info`) object to [`h5open`](@ref).
+For instance:
+
+```julia
+comm = MPI.COMM_WORLD
+info = MPI.Info()
+ff = h5open(filename, "w", comm, info)
+```
+
+MPI-distributed data is typically written by first creating a dataset
+describing the global dimensions of the data.
+The following example writes a `10 × Nproc` array distributed over `Nproc` MPI
+processes.
+
+```julia
+Nproc = MPI.Comm_size(comm)
+myrank = MPI.Comm_rank(comm)
+M = 10
+A = fill(myrank, M)  # local data
+dims = (M, Nproc)    # dimensions of global data
+
+# Create dataset
+dset = d_create(ff, "/data", datatype(eltype(A)), dataspace(dims))
+
+# Write local data
+dset[:, myrank + 1] = A
```

Comment:
Why is ` +1 ` always passed when dealing with rank? Should the library automatically add think?

Reply:
That's an issue coming from the MPI side, and I think there's not much we can do about it.

In MPI.jl, the process ranks returned by `MPI.Comm_rank` are in `0:Nproc-1`. Although this is not the most natural choice in Julia, it is consistent with the MPI C and Fortran APIs, and MPI.jl users are usually aware of this.

From the HDF5 side, we can't assume that the index passed to `setindex!` is an MPI rank, as it could be anything else.